Improvements to build sites configuration

jmchilton opened the pull request:
- This fixes #3494 so that usegalaxy.org can simply use the default sample datatypes configuration file.
- This also prevents the sample shared data files from being copied - which should be beneficial for #1472.

Ping @natefoo @martenson

jmchilton committed:
Pull build_sites out of datatypes conf.

So deployers may modify these paths and stil use the default datatypes config.

This should speed up set_metadata also since each of these config files won't be loaded as part of that.

nsoranzo commented on lib/galaxy/datatypes/registry.py:
```diff
@@ -314,23 +319,49 @@ def append_to_sniff_order():
+                return
+
+            site_type = build_site_config.get( 'type' )
+            file = build_site_config.get( 'file' )
```
It may be a good time to use a variable name which is not a Python2 reserved word.

jmchilton replied:
Good point - done!

nsoranzo commented:
Framework tests were failing, I suppose that's related: https://jenkins.galaxyproject.org/job/docker-framework/6406/

natefoo commented on scripts/common_startup.sh:
```diff
@@ -36,11 +36,6 @@ SAMPLES="
-    tool-data/shared/ucsc/manual_builds.txt.sample
-    tool-data/shared/ucsc/ucsc_build_sites.txt.sample
-    tool-data/shared/igv/igv_build_sites.txt.sample
-    tool-data/shared/rviewer/rviewer_build_sites.txt.sample
```
+:100: for getting rid of this mess.

jmchilton commented:
Turns out the builds are different than the build sites and I removed too many things from the common startup's list of file copy actions. I have fixed that and just merged everything into one commit. Hopefully the tests pass now.

martenson commented:
Thanks @jmchilton and @nsoranzo !
